### DISCUSSION TITLE
SQLite的文艺复兴 · BMPI

### DISCUSSION THREAD

**Discussion by giscus** [bmpi-dev/bmpi.dev#135] [Announcements]
# SQLite的文艺复兴 · 构建我的被动收入

本文介绍史上最流行的数据库SQLite从进程内到分布式的的文艺复兴故事。

https://www.bmpi.dev/dev/renaissance-sqlite/

**Comment by madawei2699** [comment]
- [Cloud Backed SQLite](https://sqlite.org/cloudsqlite/doc/trunk/www/index.wiki): 官方支持的云端SQLite，支持Azure和GCP，可以通过storage client直接读取或写入数据库，无需下载整个数据库；
- [SQLite HC-tree](https://sqlite.org/hctree/doc/hctree/doc/hctree/index.html): 官方的一个fork版本，使用新的数据存储结构HC-tree替换老的B-tree，能实现多写多读、分布式节点复制与更大的数据库大小限制；

**Reply by madawei2699** [reply]
SQLite HC-tree 的设计文档：https://sqlite.org/hctree/doc/hctree/doc/hctree/design.wiki 看起来挺有意思的，性能测试也很好：https://sqlite.org/hctree/doc/hctree/doc/hctree/threadtest.wiki

看起来 HC-tree 是个值得期待的大功能。

**Comment by madawei2699** [comment]
SQLite 相关好文推荐

**Reply by madawei2699** [reply]
[Why SQLite? Why Now?](https://tantaman.com/2022-08-23-why-sqlite-why-now.html)

**Comment by dreamhunter2333** [comment]
cloudflare  D1 也是基于 sqlite
https://developers.cloudflare.com/d1/

**Comment by madawei2699** [comment]
- [Zero-latency SQLite storage in every Durable Object](https://blog.cloudflare.com/sqlite-in-durable-objects/)
